File: connectors/connector-redmine/src/test/scala/com/taskadapter/connector/redmine/RedmineFactoryTest.scala
package com.taskadapter.connector.redmine

import com.taskadapter.model.AllFields
import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.{FunSpec, Matchers}

@RunWith(classOf[JUnitRunner])
class RedmineFactoryTest extends FunSpec with Matchers {
  describe("default fields") {
    it("have Assignee") {
      new RedmineFactory().getDefaultFieldsForNewConfig should contain(AllFields.assigneeLoginName)
    }
    it("skip optional ones") {
      new RedmineFactory().getDefaultFieldsForNewConfig should contain noneOf(AllFields.updatedOn,  AllFields.id,  AllFields.key)
    }
  }
}

File: connectors/connector-redmine/src/test/java/com/taskadapter/connector/redmine/RedmineFactoryTest.java
package com.taskadapter.connector.redmine;

import com.taskadapter.model.AllFields;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class RedmineFactoryTest {
    @Test
    public void defaultFieldsHaveAssignee() {
        assertThat(new RedmineFactory().getDefaultFieldsForNewConfig()).contains(AllFields.assigneeLoginName);
    }

    @Test
    public void defaultFieldsDoNotHaveOptionalFields() {
        assertThat(new RedmineFactory().getDefaultFieldsForNewConfig())
                .doesNotContain(AllFields.updatedOn, AllFields.id, AllFields.key);
    }
}
